### DISCUSSION TITLE
how to delete object with isarlink and links?

### DISCUSSION THREAD

**Discussion by PrzemyslawPluszowy** [isar/isar#1517] [Help]
hello guys, i have coloection with multiple israrlinks , i need to delete object wit
how to delete object with isarlink and links?  

  Future<void> deleteFormById(int id) async {
    final isar = await db;
    await isar.writeTxn(() async {
      isar.formModels.delete(id);
    });
  }

**Comment by Jtplouffe** [comment]
You will need to load the links before deleting the main object.

It could look something like this:

```dart
Future<void> deleteFormById(int id) async {
  final isar = await db;
  final form = await isar.formModels.get(id);
  if (form == null) {
    return;
  }

  await isar.writeTxn(() async {
    if (form.myLink.value case final myLink?) {
      await isar.myLinks.delete(myLink.id);
    }

    await isar.formModels.delete(id);
  });
}
```
